libs/parallelism/algorithms/include/hpx/parallel/algorithms/replace.hpp

```diff
@@ -652,9 +671,239 @@ namespace hpx { namespace parallel { inline namespace v1 {
-            detail::replace_copy_if<std::pair<FwdIter1, FwdIter2>>().call(
-                std::forward<ExPolicy>(policy), is_seq(), first, last, dest,
-                std::forward<F>(f), new_value, std::forward<Proj>(proj)));
+        return detail::replace_copy_if<
+            util::in_out_result<FwdIter1, FwdIter2>>()
+            .call(std::forward<ExPolicy>(policy), is_seq(), first, last, dest,
+                std::forward<F>(f), new_value, std::forward<Proj>(proj));
     }
 }}}    // namespace hpx::parallel::v1
+
+namespace hpx {
+    ///////////////////////////////////////////////////////////////////////////
+    // CPO for hpx::replace_if
+    HPX_INLINE_CONSTEXPR_VARIABLE struct replace_if_t final
+      : hpx::functional::tag<replace_if_t>
+    {
+        // clang-format off
+        template <typename Iter,
+            typename Pred, typename T, HPX_CONCEPT_REQUIRES_(
+                hpx::traits::is_iterator<Iter>::value &&
+                hpx::is_invocable_v<Pred,
+                    typename std::iterator_traits<Iter>::value_type
+                >
+            )>
+        // clang-format on
+        friend void tag_invoke(hpx::replace_if_t, Iter first, Iter last,
+            Pred&& pred, T const& new_value)
+        {
+            static_assert((hpx::traits::is_input_iterator<Iter>::value),
+                "Required at least input iterator.");
+
+            hpx::parallel::v1::detail::replace_if<Iter>().call(
+                hpx::execution::sequenced_policy{}, std::true_type{}, first,
+                last, std::forward<Pred>(pred), new_value,
+                hpx::parallel::util::projection_identity());
+        }
+
+        // clang-format off
+        template <typename ExPolicy, typename FwdIter,
+            typename Pred,  typename T, HPX_CONCEPT_REQUIRES_(
+                parallel::execution::is_execution_policy<ExPolicy>::value &&
+                hpx::traits::is_iterator<FwdIter>::value &&
+                hpx::is_invocable_v<Pred,
+                    typename std::iterator_traits<FwdIter>::value_type
+                >
+            )>
+        // clang-format on
+        friend typename parallel::util::detail::algorithm_result<ExPolicy,
+            void>::type
+        tag_invoke(hpx::replace_if_t, ExPolicy&& policy, FwdIter first,
+            FwdIter last, Pred&& pred, T const& new_value)
+        {
+            static_assert((hpx::traits::is_forward_iterator<FwdIter>::value),
+                "Required at least forward iterator.");
+
+            typedef hpx::is_sequenced_execution_policy<ExPolicy> is_seq;
+
+            return parallel::util::detail::algorithm_result<ExPolicy>::get(
+                hpx::parallel::v1::detail::replace_if<FwdIter>().call(
+                    std::forward<ExPolicy>(policy), is_seq(), first, last,
+                    std::forward<Pred>(pred), new_value,
+                    hpx::parallel::util::projection_identity()));
```

Comment:
Could we just write (I'm not sure, however):
```suggestion
            return hpx::parallel::v1::detail::replace_if<void>().call(
                    std::forward<ExPolicy>(policy), is_seq(), first, last,
                    std::forward<Pred>(pred), new_value,
                    hpx::parallel::util::projection_identity());
```